### ISSUE TITLE
[Question] NER在训练的过程中开始acc就很高，迭代后valid的acc骤降，能帮忙分析分析吗？

### ISSUE THREAD

**Issue by xmy7216** [BrikerMan/Kashgari#208]
**You must follow the issue template and provide as much information as possible. otherwise, this issue will be closed.
请按照 issue 模板要求填写信息。如果没有按照 issue 模板填写，将会忽略并关闭这个 issue**

## Check List
Thanks for considering to open an issue. Before you submit your issue, please confirm these boxes are checked.

**You can post pictures, but if specific text or code is required to reproduce the issue, please provide the text in a plain text format for easy copy/paste.**

-  I have searched in [existing issues](https://github.com/BrikerMan/Kashgari/issues?utf8=%E2%9C%93&q=is%3Aissue+) but did not find the same one.
-  I have read the [documents](https://kashgari.bmio.net)

## Environment

- OS [e.g. Mac OS, Linux]:Linux
- Python Version:3.6.9
- requirements.txt:

```txt
[Paste requirements.txt file here]
```

## Question
我使用自己的数据集训练测试NER，有4种实体类型，每一种类型有7000条train，valid和test各1500条，主体代码如下：
```
import kashgari
kashgari.config.use_cudnn_cell = True

from kashgari.embeddings import BERTEmbedding
embedding = BERTEmbedding(task = kashgari.LABELING, model_folder = './chinese_L-12_H-768_A-12', sequence_length = 100)

from kashgari.tasks.labeling import BiLSTM_CRF_Model
model = BiLSTM_CRF_Model(embedding)
model.fit(train_x,
          train_y,
          x_validate = valid_x,
          y_validate = valid_y,
          epochs = 50,
          batch_size = 128)

model.evaluate(test_x, test_y)
```

但是训练开始acc就很高，迭代后valid的acc骤降，然后loss一直在降低，但acc也在降低，不知道是什么原因？
![image](https://user-images.githubusercontent.com/31274050/63327671-3d590900-c361-11e9-8a6e-e4c1235bcf3c.png)

[A clear and concise description of what you want to know.]

**Comment by BrikerMan** [comment]
先试试 BiLSTM_Model 看看会怎么样

**Comment by xmy7216** [comment]
> 先试试 BiLSTM_Model 看看会怎么样

谢谢，根据您的建议，换成了BiLSTM_Model，也使用了BERT，epoch设置为15，其他参数未调整，训练的初始正确率和loss感觉还是太好了，
![image](https://user-images.githubusercontent.com/31274050/63344784-d00a9f80-c383-11e9-8ad6-423faa1c90b2.png)
但是在测试集上的结果还不如上面不正常的BILSTM_CRF模型

**Comment by BrikerMan** [comment]
似乎是 CRF 层的 loss 和 acc 计算有问题，如果你训练完 CRF，用 evaluate 函数测试效果怎么样

**Comment by xmy7216** [comment]
测试集的f1得分在70%左右







------------------ 原始邮件 ------------------
发件人: Eliyar Eziz <notifications@github.com>
发送时间: 2019年8月20日 22:17
收件人: BrikerMan/Kashgari <Kashgari@noreply.github.com>
抄送: xmy7216 <1285074029@qq.com>, Author <author@noreply.github.com>
主题: 回复：[BrikerMan/Kashgari] [Question] NER在训练的过程中开始acc就很高，迭代后valid的acc骤降，能帮忙分析分析吗？ (#208)




似乎是 CRF 层的 loss 和 acc 计算有问题，如果你训练完 CRF，用 evaluate 函数测试效果怎么样

**Comment by BrikerMan** [comment]
我的意思，CRF 和非 CRF 模型的 F1 各自怎么样？

**Comment by xmy7216** [comment]
> 我的意思，CRF 和非 CRF 模型的 F1 各自怎么样？

BiLSTM_CRF_Model训练50epoch后在test集上的表现：
![image](https://user-images.githubusercontent.com/31274050/63423050-14f30c80-c43e-11e9-8514-98a09c5cfcb0.png)
BiLSTM_Model训练15epoch后在test集上的表现：
![image](https://user-images.githubusercontent.com/31274050/63425598-808ba880-c443-11e9-85fa-402281ccce1a.png)

**Comment by BrikerMan** [comment]
那就是说 CRF 层实际表现好一些，但是 loss 显示有问题对吧？

**Comment by BrikerMan** [comment]
麻烦尝试一下把数据集使用 bert-tokenizer 进行分词，参考  https://kashgari.bmio.net/embeddings/bert-embedding/#example-usage-text-classification

**Comment by zxyza** [comment]
我也发现了这个问题，但是我感觉应该是标注数据不够准确的原因吧，比如 我的train_data中有电子计算机 ，然后dev_data中存在  已经被人工标注的数据例如  电子计算器，训练后的模型针对  电子计算器进行预测的时候只能将  计算器  标注出来，但是  电子  却没有被预测标注出来，这种二者预测结果和标注的结果并不完全一致的时候就会导致正确率下降对吧；

**Comment by harry771** [comment]
有一个关于ner fine-tune的问题，就是预测的时候该调用哪个函数，fine-tune之后要测一下效果，是直接调用predict吗？
